.github/workflows/ios-build.yml

```diff
@@ -2,38 +2,78 @@ name: Test iOS build
 on:
   pull_request:
     paths:
+      - '.github/workflows/ios-build.yml'
+      - 'RNReanimated.podspec'
       - 'ios/**'
       - 'Common/**'
+      - 'Example/ios/**'
+      - 'FabricExample/ios/**'
   push:
     branches:
       - main
 
 jobs:
   build:
-    # runs-on: macos-latest // issue: https://github.com/actions/virtual-environments/issues/4060
-    runs-on: macos-11
+    runs-on: macos-12
     strategy:
       matrix:
         working-directory: [Example, FabricExample]
     concurrency:
-      group: ios-${{ github.ref }}
-      cancel-in-progress: true
+      group: ios-${{ matrix.working-directory }}-${{ github.ref }}
     steps:
-      - name: checkout
+      - name: Check out Git repository
         uses: actions/checkout@v2
-      - name: Use Node.js 14
-        uses: actions/setup-node@v2
+
+      - name: Restore Reanimated node_modules from cache
+        uses: actions/cache@v2
+        id: cache-node-modules-reanimated
+        with:
+          path: |
+            node_modules
+          key: ${{ runner.os }}-node-modules-reanimated-${{ hashFiles('yarn.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-node-modules-reanimated
+
+      - name: Install Reanimated node_modules
+        run: yarn install --frozen-lockfile
```

Comment:
Why do you use `--frozen-lockfile` option? Does this mean CI will [fail](https://classic.yarnpkg.com/lang/en/docs/cli/install/#toc-yarn-install-frozen-lockfile) when we change some dependency in `package.json`?

Reply:
I've seen this flag being used in some other CI job in other repository. Assume that we made some changes in `package.json` but forgot to commit `yarn.lock`. Now, when CI installs node_modules, it will generate (update) `yarn.lock` in the CI runner filesystem. This flag means that `yarn install` will fail in such case. For my understanding, it can be used to detect differences between `package.json` and `yarn.lock`.

Reply:
Ok, I see it now, thanks!